src/compiler/scala/tools/nsc/transform/async/AsyncPhase.scala

```diff
@@ -123,14 +123,17 @@ abstract class AsyncPhase extends Transform with TypingTransformers with AnfTran
           }
 
           atOwner(dd, dd.symbol) {
-            val trSym = dd.vparamss.head.head.symbol
+            val trSym = dd.vparamss.head.last.symbol
+            val selfSym = if (dd.symbol.owner.isTerm) dd.vparamss.head.head.symbol else NoSymbol
```

Comment:
This is the condition to tell whether the async method is a sibling or a member, right?

I see that in the "classic" transformation the owner is the state machine class. How do we know that in the "sibling" transformation, the owner will always be a term?

Reply:
The assumption is that it is the state machine class and impl method are enclosed in the place where the original async-ed expression was located, which is always somewhere term-owned.